yunwoo/programmers-0/배열-만들기.js

```diff
@@ -0,0 +1,15 @@
+function solution(arr) {
+  const stk = [];
+  let i = 0;
+
+  while (i <= arr.length - 1) {
+    if (!stk.length) {
+      stk.push(arr[i++]);
+      continue;
+    }
+
+    stk.at(-1) >= arr[i] ? stk.pop() : stk.push(arr[i++]);
+  }
```

Comment:
```suggestion
  while (i <= arr.length - 1) {
    const shouldAppendLastItem = !stk.length || stk.at(-1) < arr[i];
    
    if (shouldAppendLastItem) {
      stk.push(arr[i++]);
      continue;
    }

    stk.pop()
  }
```
- push 부분이 공통적이라 조건식을 함께 사용하면 좀 더 복잡도 높은 삼항 연산자와 중복 코드를 제거 할 수 있을 것 같습니다!

Reply:
- 헉 저런 방법은 생각도 못했습니다 훨씬 깔끔하네요!! 감사합니다 👍